streamlit_app.py

```diff
@@ -183,21 +180,21 @@ def get_random_features(feature_names, seed):
 
 # Hash the TensorFlow session, the pg-GAN model, and the TL-GAN model by id
 # to avoid expensive or illegal computations.
-@st.cache(show_spinner=False, hash_funcs=TL_GAN_HASH_FUNCS)
-def generate_image(session, pg_gan_model, tl_gan_model, features, feature_names):
+@st.experimental_memo(show_spinner=False, ttl=60*60)
```

Comment:
Is the `ttl` here to prevent memory issues or any other reason?

Reply:
If only for memory, I think we should set it to one day (i.e. `24*60*60`) to make it faster for most people while still clearing memory regularly.

Reply:
That was a setting intended to prevent memory issues. It used to be `None` within `@st.cache`, meaning cache entries would not expire. I agree with you, let's have it longer, a day looks reasonable.

```suggestion
@st.experimental_memo(show_spinner=False, ttl=24*60*60)
```